### PR TITLE
feat(kotlin): update client, add readme

### PR DESCRIPTION
**Pull request by Haroenv**
updated

### CODE REVIEW

File: kotlin/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+First, build the project
+
+```shell
+./gradlew build
+```
+
+Then, run the project
+
+```shell
+??????????
```
Comment (Haroenv) [inline kotlin/README.md:28]: not sure, I didn't get that far

Comment (Aallam) [reply kotlin/README.md]: I've updated the guide, please take a look

File: kotlin/bin/main/App.kt [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import com.algolia.search.client.ClientSearch
```
Comment (Haroenv) [inline kotlin/bin/main/App.kt:1]: you added this file @Aallam, is that expected?

**Review by Haroenv — COMMENTED** [review]
it works for me now with the steps in the readme

File: kotlin/bin/main/App.kt (hunk 2 above)
Comment (Aallam) [reply kotlin/bin/main/App.kt:1]: I renamed the file from `app.kt` to `App.kt`, to match the convention, and this line:
```kotlin
application {
     // Define the main class for the application.
     mainClass = 'AppKt'
 }
```

Comment (Haroenv) [reply kotlin/bin/main/App.kt:1]: I mean that there's a bin/App.kt and src/App.kt (both tracked) @Aallam

Comment (Aallam) [reply kotlin/bin/main/App.kt]: Oh you are right, I removed the file under `bin/` and added an `.gitignore` file

**Review by Haroenv — COMMENTED** [review]
can't approve my own PR, but as @Aallam made it work, and it works for me, so let's merge

**Pull request merged**